openMINDS_pipeline/resolver.py

```diff
@@ -150,6 +165,40 @@ def _do_resolve_categories(schema: SchemaStructure, schemas_by_category):
                         embedded_types.extend(schemas_by_category[embedded_category])
                 schema_payload["properties"][p][TEMPLATE_PROPERTY_EMBEDDED_TYPES] = sorted(embedded_types)
                 del schema_payload["properties"][p][TEMPLATE_PROPERTY_EMBEDDED_CATEGORIES]
+
+            # Write namespace for '_linkedTyped' and '_embeddedType'
+            if version == "latest" or float(version[1:]) >= 4:
+                if TEMPLATE_PROPERTY_LINKED_TYPES in schema_payload["properties"][p]:
+                    schema_payload["properties"][p][TEMPLATE_PROPERTY_LINKED_TYPES] = [
+                        schema.namespaces['types'] + _type.split(":")[-1].split("/")[-1] if (_type.__contains__(
```

Comment:
This list comprehension is very hard to understand, and reoccurs a few lines down for embedded types. I suggest breaking this out into a function, and perhaps assigning `_type.split(":")[-1].split("/")[-1]` to a variable with a meaningful name; at a minimum more comments are needed.

Also, `_type.__contains__(x) == True` should be written as `x in _type`.

Reply:
I've simplified the logic and removed unnecessary code, it should be better now.